cni/network/network.go

```diff
@@ -569,14 +590,14 @@ func (plugin *NetPlugin) Add(args *cniSkel.CmdArgs) error {
 		}
 
 		var epInfo network.EndpointInfo
-		epInfo, err = plugin.createEndpointInternal(&createEndpointInternalOpt)
+		epInfo, err = plugin.createEndpointInternal(&createEndpointInternalOpt, ifIndex)
```

Comment:
maybe we can pass interface info instead of indexing everywhere

Reply:
The only reason why I am doing it this way is for performance. We have to index for infra nic in multiple locations otherwise. 

As stated in earlier conversations, having to loop through the interfaceInfos and passing index could be removed if we change it from an array into a map. I am saving that for another PR as the scope for this one is just to remove `defaultInterfaceInfo` and `secondayInterfaceInfos`